common.go

```diff
@@ -25,6 +25,31 @@ func MarkdownEscape(s string) string {
 	return "```\n" + s + "\n```\n"
 }
 
+// NetworkToEndpoint maps network names to their corresponding StreamingFast endpoints
+func NetworkToEndpoint(network string) string {
+	endpoints := map[string]string{
+		"mainnet":   "mainnet.eth.streamingfast.io:443",
+		"bnb":       "bnb.streamingfast.io:443",
+		"bsc":       "bnb.streamingfast.io:443",
+		"polygon":   "polygon.streamingfast.io:443",
+		"arbitrum":  "arbitrum.streamingfast.io:443",
+		"optimism":  "optimism.streamingfast.io:443",
+		"base":      "base.streamingfast.io:443",
+		"avalanche": "avalanche.streamingfast.io:443",
+		"fantom":    "fantom.streamingfast.io:443",
+		"amoy":      "polygon.streamingfast.io:443", // Polygon testnet uses same endpoint
+		"holesky":   "holesky.eth.streamingfast.io:443",
+		"sepolia":   "sepolia.eth.streamingfast.io:443",
+	}
+	
+	if endpoint, exists := endpoints[network]; exists {
+		return endpoint
+	}
+	
+	// Fallback to a generic pattern if not found
+	return network + ".streamingfast.io:443"
+}
+
```

Comment:
We have a library https://github.com/streamingfast/firehose-networks that does all this heavy lifting. You can use https://github.com/streamingfast/firehose-networks/blob/master/REFERENCE.md#getsubstreamsendpointkey-string to avoid having to hardcode a list like that.

The thing however is that the library caches the latest registry, since this project runs in a server, it could mean not updating the latest registry until next restart of the pod which can take a long time. Ensure that in the server's main starting point/when the application launches the server, make sure to do https://github.com/streamingfast/firehose-networks/blob/master/REFERENCE.md#scheduleupdatelatestregistryctx-contextcontext-interval-timeduration-logger-zaplogger, let's schedule to update very 12 hours.